pkg/customresources/keymanagementproviders/api.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keymanagementproviders
+
+import (
+	"crypto"
+	"crypto/x509"
+
+	kmp "github.com/deislabs/ratify/pkg/keymanagementprovider"
+)
+
+// KMPCertManager is an interface that defines the methods for managing certificate stores across different scopes.
+type KMPCertManager interface {
```

Comment:
curious why is the Cert Manager and Key Manger separated?

Reply:
just because Get and Add operations have different input/output, one is publicKey and the other is Certificate. If I use `interface` to uniform the input/output, it might complicate the implementation.

But if we remove the extra namespace mapping, we could remove the interfaces.